File: bindings/kotlin/examples/OwnedObjects.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.Address
import iota_sdk.GraphQlClient
import iota_sdk.ObjectFilter
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {
        val client = GraphQlClient.newTestnet()
        val address = Address.zero()
        val objectFilter = ObjectFilter(owner = address)
        val objectsPage = client.objects(objectFilter)
        println("Owned objects (${objectsPage.data.size}):")
        for (obj in objectsPage.data) {
            println(obj.id().toHex())
        }
    } catch (e: Exception) {
        e.printStackTrace()
        kotlin.system.exitProcess(1)
    }
}

File: bindings/swift/examples/OwnedObjects.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import IotaSDK

@main
struct OwnedObjectsExample {
  static func main() async throws {
    let client = GraphQlClient.newTestnet()
    let address = Address.zero()
    let objectsPage = try await client.objects(filter: ObjectFilter(owner: address))
    print("Owned objects(\(objectsPage.data.count)):")
    for obj in objectsPage.data {
      print(obj.id().toHex())
    }
  }
}
